client/go/outline/config/config_iptable.go

```diff
@@ -0,0 +1,162 @@
+	"context"
+	"errors"
+	"fmt"
+	"net/netip"
+
+	"github.com/Jigsaw-Code/outline-apps/client/go/configyaml"
+	"github.com/Jigsaw-Code/outline-apps/client/go/outline/iptable"
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+)
+
+type ipTableEntryConfig struct {
+	IP     string                `yaml:"ip,omitempty"`
+	Dialer configyaml.ConfigNode `yaml:"dialer"`
+}
+
+type ipTableRootConfig struct {
+	Table []ipTableEntryConfig `yaml:"table"`
+}
+
+// parsedIPTableStreamEntry is an internal struct to hold a parsed prefix and its corresponding stream dialer.
+type parsedIPTableStreamEntry struct {
+	prefix netip.Prefix
+	dialer transport.StreamDialer
+}
+
+func parseIPTableStreamDialer(
+	ctx context.Context,
+	configMap map[string]any,
+	parseSD configyaml.ParseFunc[*Dialer[transport.StreamConn]],
+) (*iptable.StreamDialer, bool, error) {
+	var rootCfg ipTableRootConfig
+	if err := configyaml.MapToAny(configMap, &rootCfg); err != nil {
+		return nil, false, fmt.Errorf("failed to map iptable stream config: %w", err)
+	}
+
+	if len(rootCfg.Table) == 0 {
+		return nil, false, errors.New("iptable config 'table' must not be empty for stream dialer")
+	}
+
+	parsedEntries := make([]parsedIPTableStreamEntry, 0, len(rootCfg.Table))
+	var defaultDialerEntry *parsedIPTableStreamEntry
+	hasDirectConn := false
+
+	for i, entryCfg := range rootCfg.Table {
+		parsedSubDialer, err := parseSD(ctx, entryCfg.Dialer)
+		if err != nil {
+			return nil, false, fmt.Errorf("failed to parse nested stream dialer for table entry %d (ip: %s): %w", i, entryCfg.IP, err)
+		}
+
+		if parsedSubDialer.ConnType == ConnTypeDirect {
+			hasDirectConn = true
+		}
+
+		currentEntry := parsedIPTableStreamEntry{
+			dialer: transport.FuncStreamDialer(parsedSubDialer.Dial),
+		}
+
+		if entryCfg.IP == "" { // Default dialer
+			if defaultDialerEntry != nil {
+				return nil, false, errors.New("multiple default dialers specified in iptable for stream")
+			}
+			defaultDialerEntry = &currentEntry
+			continue
+		}
+
+		var prefix netip.Prefix
+		parsedPrefix, errPrefix := netip.ParsePrefix(entryCfg.IP)
+		if errPrefix == nil {
+			prefix = parsedPrefix
+		} else {
+			addr, errAddr := netip.ParseAddr(entryCfg.IP)
+			if errAddr != nil {
+				return nil, false, fmt.Errorf("iptable entry %d IP '%s' is not a valid IP address or CIDR prefix: failed to parse as prefix (%v) and failed to parse as address (%v)", i, entryCfg.IP, errPrefix, errAddr)
+			}
+			prefix = netip.PrefixFrom(addr, addr.BitLen())
+		}
+
+		currentEntry.prefix = prefix
+		parsedEntries = append(parsedEntries, currentEntry)
+	}
+
+	table := iptable.NewIPTable[transport.StreamDialer]()
+
+	for _, entry := range parsedEntries {
+		table.AddPrefix(entry.prefix, entry.dialer)
+	}
+
+	dialer, err := iptable.NewStreamDialer(table)
+
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to create IPTableStreamDialer: %w", err)
+	}
+
+	isFullTunnel := !hasDirectConn && (defaultDialerEntry != nil || isExhaustive(parsedEntries))
```

Comment:
isFullTunnel is a boolean, it doesn't capture the 3 states we need.

We need two booleans: isAllDirect and IsAllTunneled.
They can both be initialised according to the default dialer, if it's set, or true if no default dialer and at least one route.

Then you can calculate the 3 states we need:
IsAllTunneled -> Tunneled
IsAllDirect -> Direct
else -> partial

Then parseIPTableStreamDialer can return the ConnType, not a boolean (or the Dialer[])